R/mobr.R

```diff
@@ -473,6 +503,18 @@ get_plot_dens = function(comm, density_stat){
     return(plot_dens)   
 }
 
+#' Auxillary function for effect_ functions
+#' Compute an overall p-value for one factor in the discrete case
+#' p-value is based on mean squared difference from zero summed across the scales
+#' @keywords internal
+get_overall_p = function(effort, deltaS_emp, deltaS_null){
+    delta_effort = (c(effort, 0) - c(0, effort))[1:length(effort)]
```

Comment:
Ok so I'm looking at this closer now and I'm not sure what the appropriate delta effort value should be for the last sampling effort. It looks like according to Loosmore (above Eq. 4) delta_effort_k = effort_k+1 - effort_k so if you're at the last effort then you cannot compute the last delta value.

Reply:
this occurred to me because `diff(effort)` actually returns a vector that is one less than the length of effort which makes sense of course.

Reply:
In the current implementation I think delta_effort has the same length as effort, and delta_effort[1] is just effort[1], or the first scale. This kind of makes sense to me, because what we are trying to compute, by multiplying delta_effort with detalS^2, is similar to a crude integration, or area under the curve across the scales. By keeping delta_effort[1] = effort[1], we are essentially saying, ok let's assume the effect (deltaS) can be approximated by deltaS[1], from the spatial scale of 0 to effort[1]. Does that sound reasonable?